.evergreen/config_generator/etc/distros.py

```diff
@@ -0,0 +1,296 @@
+from typing import Optional
+
+from pydantic import BaseModel, validator
+from packaging.version import Version
+
+
+class Distro(BaseModel):
+    """
+    Defines common properties of a given Evergreen distro.
+
+    * name: Name of the distro.
+    * os: Name of the operating system.
+    * os_type: One of Linux, MacOS, or Windows.
+    * os_ver: Version of the operating system.
+    * vs_ver: Version of Visual Studio available.
+    * size: Size of tasks the distro is designed to handle.
+    * arch: Target architecture.
+    """
+
+    name: str
+    os: Optional[str]
+    os_type: Optional[str]
+    os_ver: Optional[str]
```

Comment:
Give defaults ` = None` to the fields that may be omitted.

Reply:
It works at runtime, but static analysis has trouble with it (e.g. pyright flags all of the Distro calls as having missing arguments)

Reply:
That is strange. PyLint does not complain about lack of `= None` for optional fields.

Since [pydantic documentation](https://docs.pydantic.dev/usage/types/#typing-iterables) also uses `= None`, I will follow your suggestion and add them in.

Reply:
Pylint is significantly weaker as a type checker than mypy or pyright. If you have vscode and the Python/Pylance extensions, I would recommend enabling strict-typechecking mode for Pylance and you will get Pyright annotations. It will be noisy for un-annotated code, but catches problems very very quickly.

Reply:
TIL. I was wondering how to get proper warnings for type annotations. It seems like PyLint wasn't sufficient for that purpose. I am seeing many more useful warnings now. I will make an effort to address them in followup PRs.